### ISSUE TITLE
Adding missing months (November, December, January, February (which would also have leap years every time year in multiple of 4 occurs, or if it is on 00, every 400 years))

### ISSUE THREAD

**Issue by Christipher** [OpenRCT2/OpenRCT2#1634]
I was wondering if you could add code to deal with the missing months of November, December, January, and February (February having leap years).

This is done because most parks run from March - October, though down in Florida that is broken and the parks run year - round (with the busiest days on Christmas and New Year's Eve); this is also so that timing can be more realistic.

Also with that would come more climates - Hot and Wet (more for Tropical Rainforest climates, but also for climates where wetlands also occur); cool and dry (for cold deserts), and the ever - lurking Island climates.

**Comment by vzybilly** [comment]
I'm not sure if I'm correct or not, but I think a year right now is exactly an hour? adding the four months would change that unless the length of a day was changed as well... leap years I don't think is really needed, most all parks I play end before or at year 4 and that would be a hefty bit of logic to handle unless there is already a really good calendar system that we could just swap in...

the skipping of the four months does still off-put me but would it be worth the change? it might also require changing whole data structures because the current month scheme fits into 3 bits perfectly allowing for simple AND operations to pull current month of year and year from a byte(63 years total)?
I don't actually know what all is done for the months and years but that's how I would do it if I was making this game back when computer memory was gold and even more so if real parks in my area did that...

**Comment by Phasedetect** [comment]
This is a duplicate of an old closed issue brought up a long time ago.

There are two categories of enhancements. The first category is useful tools and changes for building and such, which all roughly can be summed up with [sandbox mode](https://github.com/OpenRCT2/OpenRCT2/wiki/Sandbox-Mode) and the [cheats window](https://github.com/OpenRCT2/OpenRCT2/wiki/Cheats-Window). These cheats, while they defeat limitations of the original game, are easy modifications to existing code and are basically anything that could be done with a trainer.

The second category is changes to game mechanics. These things aren't going to even be considered until all code is reversed, because we can't make changes to the game when original functions are still being called.

Although this is a much-requested feature, it won't happen any time soon.

**Comment by Gymnasiast** [comment]
It has been suggested once before. It's not what I would call much-requested.

Anyway, there is a very good reason for it, which you already mentioned actually: most parks are closed in the winter months, as most roller coasters cannot operate properly when it's too cold, and maintenance work is carried out during those months. Of course, there are some exceptions, even in temperate climates like in The Netherlands.

However, it would complicate things beyond belief. An RCT year (8 months) takes about 90 minutes. It would either mean most of your rides will be closed for 45 minutes, or open and close every few minutes (a day takes about 20-40 seconds in RCT!) to take the weather in account. Of course, in areas where it's always warm enough, that wouldn't apply, but then it would mean that you end up with a seriously complicated game, with very inconsistent behaviour between climates.

In short, I am against this. It's far too complicating and doesn't improve gameplay - in fact, I would say it degrades gameplay for the reasons I mentioned above.

**Comment by vzybilly** [comment]
All that extra work would take from the game but what about just adding the months and call it done? year becomes longer but nothing else is changed, no weather or anything so the rides can all stay open year round, would that mess with things to much?

**Comment by IntelOrca** [comment]
What is more feasible is saying which 8 of the 12 months should be used. So if your park was in the southern hemisphere, you would have the opposite side of the year.

**Comment by Gymnasiast** [comment]
@vzybilly : Because that takes realism out of the game and will interfere with the mechanics, because the year will be 1.5 times longer. Unless you make the days shorter. But then you could ask yourself what purpose it would serve. In my opinion: it solves _absolutely nothing_.

@IntelOrca : Sounds like a far better solution to me.

**Comment by vzybilly** [comment]
while it doesn't solve anything and I don't really go either way, the 8 out of 12 messes with my mind and I get confused over it but I'm pretty sure that's more of a fault in the american education system than anything...

**Comment by Patrik356b** [comment]
This would be awkward for the occasional player

**Comment by ipatix** [comment]
Well, I personally always wondered about that but it kinda made sense in terms of parks closing down throughout the colder months. I wouldn't consider it being akward at all.
Maybe you could make this adjustable some time in the future but for now I would call it something really low priority.
